lib/Bio/EnsEMBL/DataCheck/Checks/CheckSyntenySanity.pm

```diff
@@ -0,0 +1,121 @@
+  GROUPS         => ['compara', 'compara_syntenies'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['dnafrag', 'dnafrag_region', 'genome_db', 'genomic_align', 'method_link', 'method_link_species_set', 'synteny_region']
+};
+
+sub tests {
+  my ($self) = @_;
+  
+  my $dba = $self->dba;
+  my $helper = $dba->dbc->sql_helper;
+  my $mlss_adap = $dba->get_MethodLinkSpeciesSetAdaptor;
+  
+  #Collect mlss_ids for synteny methods
+  my $mlss = $mlss_adap->fetch_all_by_method_link_type("SYNTENY");
+
+  foreach my $mlss ( @$mlss ) {
+    my $mlss_id = $mlss->dbID;
+    
+    my $gab_mlss_list = $mlss->get_all_sister_mlss_by_class('GenomicAlignBlock.pairwise_alignment');
+    
+    my $gdb_ids = $mlss->species_set->genome_dbs;
+        
+    #Collect dnafrag_ids longer than 1Mb with exceptions
+    foreach my $gdb ( @$gdb_ids ) {
+      my $gdb_id = $gdb->dbID;
+      my $dnafrag_sql = qq/
+      SELECT dnafrag_id 
+        FROM dnafrag 
+      WHERE genome_db_id = $gdb_id
+        AND cellular_component!='MT' 
+        AND length > 1000000
+      /;
+
+      my $dnafrag_ids = $helper->execute_simple( -SQL => $dnafrag_sql );
+      
+      #Check for reasonable count of synteny regions, fine if more than none
+      foreach my $dnafrag_id ( @$dnafrag_ids ) {
+        my $synteny_sql = qq/
+          SELECT COUNT(*) 
+            FROM synteny_region
+              JOIN dnafrag_region USING (synteny_region_id) 
+          WHERE method_link_species_set_id = $mlss_id
+            AND dnafrag_id = $dnafrag_id
+        /;
+        
+        #If no synteny regions counted, check genomic_alignment_blocks
+        my $synteny_count = $helper->execute_single_result( -SQL => $synteny_sql );
+        my $alignment_count = 0;
+        if ( $synteny_count == 0 ) {
+          foreach my $gab_mlss ( @$gab_mlss_list ) {
+            #print Dumper($gab_mlss);
+            my $gab_mlss_id = $gab_mlss->method->dbID;
+            my $dnafrag_count_sql = qq/
+              SELECT ga2.dnafrag_id, count(*) as count 
+                FROM genomic_align ga1 
+                  JOIN genomic_align ga2 USING (genomic_align_block_id)
+              WHERE ga1.dnafrag_id = $dnafrag_id
+                AND ga1.method_link_species_set_id = $gab_mlss_id
+                AND ga1.dnafrag_id <> ga2.dnafrag_id 
+              GROUP BY ga2.dnafrag_id
+              ORDER BY count(*) DESC LIMIT 1
+            /;
+            my $aln_array = $helper->execute(  
+              -SQL => $dnafrag_count_sql, 
+              -USE_HASHREFS => 1,
+              -CALLBACK     => sub {
+                my $row = shift @_;
+                return { dnafrag_id => $row->{dnafrag_id}, count => $row->{count} };
+              },
+            );
+            if ( ( scalar @$aln_array > 0 ) && ( @$aln_array[0]->{count} > $alignment_count ) ) {
+              $alignment_count = @$aln_array[0]->{count};
```

Comment:
Do the `cmp_ok` here so that you can add the other dnafrag_id to the error message. 
The algorithm would be simpler: for every alignment mlss_id, count the highest number of of blocks the dnafrag has against another dnafrag and complain if that is more than 1000.
You wouldn't need alignment_count any more